fortress/install_osx_src.md

```diff
@@ -0,0 +1,245 @@
+
+Add `osrf/simulation` to Homebrew formulae
+
+```bash
+brew update
+brew tap osrf/simulation
+```
+
+Install all dependencies:
+
+Dependency for Ogre:
+
+```bash
+brew cask install xquartz
+```
+
+General dependencies:
+
+```bash
+brew install assimp boost bullet cmake cppzmq dartsim@6.10.0 doxygen eigen fcl ffmpeg flann freeimage freetype gflags google-benchmark gts ipopt irrlicht jsoncpp libccd libyaml libzzip libzip nlopt ode open-scene-graph ossp-uuid ogre1.9 ogre2.1 pkg-config protobuf qt qwt rapidjson ruby tbb tinyxml tinyxml2 urdfdom zeromq
+```
+
+`dartsim@6.10.0` and `qt5` are not sym-linked. To use those dependencies when building
+`ignition-physics2` and `ignition-gui3`, run the following after installation to add them to `/use/local`:
+
+```bash
+# dartsim@6.10.0
+export CMAKE_PREFIX_PATH=${CMAKE_PREFIX_PATH}:/usr/local/opt/dartsim@6.10.0
+export DYLD_FALLBACK_LIBRARY_PATH=${DYLD_FALLBACK_LIBRARY_PATH}:/usr/local/opt/dartsim@6.10.0/lib:/usr/local/opt/octomap/local
+export PKG_CONFIG_PATH=${PKG_CONFIG_PATH}:/usr/local/opt/dartsim@6.10.0/lib/pkgconfig
+# qt5
+export CMAKE_PREFIX_PATH=${CMAKE_PREFIX_PATH}:/usr/local/opt/qt
+```
+
+### Install compiler requirements
+
+The Ignition Libraries require the Xcode 10 compiler on MacOS Mojave.
+
+On Mac machines, gcc is acquired by installing Xcode command line tools.
+The required version of Xcode for Fortress is Xcode 10.3, which can be downloaded from
+[Apple Developer Site](https://developer.apple.com/download/more/).
+You will need to sign in to your Apple account and download the Mojave version of
+Xcode command line tools. Command line tools can also be obtained by downloading
+Xcode from the Apple App Store (installing the full app may take over an hour).
+
+## Building the Ignition Libraries in MacOS Catalina (10.15)
+
+If you want to compile Ignition Libraries in MacOS Catalina (10.15) you will need to apply some patches in your filesystem:
+
+ - `/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.15.sdk/System/Library/Frameworks/Ruby.framework/Headers/ruby/ruby/intern.h`
+
+Create a file called `intern.patch` with the following content:
+
+```patch
+--- intern.h    2019-12-16 18:17:08.000000000 +0100
++++ /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.15.sdk/System/Library/Frameworks/Ruby.framework/Headers/ruby/ruby/intern.h
+@@ -14,6 +14,10 @@
+ #ifndef RUBY_INTERN_H
+ #define RUBY_INTERN_H 1
+
++#if __cplusplus > 199711L
++#define register      // Deprecated in C++11.
++#endif  // #if __cplusplus > 199711L
++
+ #if defined(__cplusplus)
+ extern "C" {
+ #if 0
+```
+
+Now we can apply the patch:
+
+```{.sh}
+sudo patch -p0 < intern.patch
+```
+
+ - `/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.15.sdk/System/Library/Frameworks/Ruby.framework/Headers/ruby/ruby/config.h`
+
+Create a file called `config.patch` with the following content:
+
+```
```

Comment:
```suggestion
```patch
```
We used `patch` for the other patch codeblock above, so it'd be good to add it here for consistency (adding this renders a colored diff in GitHub, which makes it easier to read for those who are looking at these files on GitHub instead of on the Ignition Robotics website)

Reply:
Done!